h/accounts/__init__.py

```diff
@@ -67,11 +46,11 @@ def authenticated_user(request):
     # an arbitrary request is NOT safe (e.g. POST requests).
     if request.authenticated_userid and not user:
         request.session.invalidate()
-        raise httpexceptions.HTTPFound(
-            location=request.current_route_url())
+        raise httpexceptions.HTTPFound(location=request.url)
```

Comment:
`current_route_url` is useful if you need to modify or adjust the current URL (either by updating the matched params or by adding/changing the query string). `request.url` is just the current URL, no magic. I've changed it just so this function is a little easier to test (see the commit message on 5572c9d).